lib/note-detail.jsx

```diff
@@ -60,6 +68,21 @@ export default React.createClass( {
 		}
 	},
 
+	onEditorChanged: function() {
+		clearTimeout( saveTimerId );
+		// Save the note after user stops typing
+		saveTimerId = delay( this.saveNote, saveDelay );
+	},
```

Comment:
I gave debounce a try but it didn't do the trick for me. It would delay the first execution, but then all of the rest of the queued calls to the functions would fire after the 3 second delay. Is that what `maxWait` can help with?